### ISSUE TITLE
Add support for vcpkg "features"

### ISSUE THREAD

**Issue by crlf0710** [mcgoo/vcpkg-rs#7]
For example, loading "harfbuzz[icu]".

**Comment by mcgoo** [comment]
Thanks for the report. Could you clarify what you mean by this?

As it is, the intention is that `vcpkg::find_package("harfbuzz").unwrap();` will find `harfbuzz` and whatever dependencies it is installed with in your vcpkg tree.  If the user has installed it with feature `icu` enabled, `icu` should get linked also. Assuming vcpkg-rs is working properly, you would just document that harfbuzz with icu is required.

Right now, if the user's vcpkg installation does not have `harfbuzz` installed, vcpkg-rs emits no metadata and returns an error.  I suppose it would be possible to specify to vcpkg-rs a required set of features and return an error if they are not all present. Is that what you are looking for?

**Comment by ras0219-msft** [comment]
To hijack the thread a bit: It might also be interesting to support an "install if missing" mode

**Comment by crlf0710** [comment]
@mcgoo 
> I suppose it would be possible to specify to vcpkg-rs a required set of features and return an error if they are not all present.

Yes, this is what is on my mind. And besides this, currently vcpkg-rs doesn't parse the status text correctly when there are features. For example, when i do this:
```
vcpkg_cli probe harfbuzz
```
I get:
```
didn't know how to deal with port :-
{"Architecture": "x64-windows", "Depends": "icu", "Description": "icu support for harfbuzz", "Feature": "icu", "Multi-Arch": "same", "Package": "harfbuzz", "Status": "install ok installed"}
didn't know how to deal with port :-
{"Architecture": "x64-windows", "Description": "Builtin (UCDN) Unicode callbacks support", "Feature": "ucdn", "Multi-Arch": "same", "Package": "harfbuzz", "Status": "install ok installed"}
```
This is annoying and apparently something is missing from the parsing part.

**Comment by mcgoo** [comment]
I just published `vcpkg` 0.2.6 which correctly handles linking libraries required bu optional features. Now `vcpkg::find_package("harfbuzz").unwrap();` will correctly link the libs for `icu` if the `icu` feature is enabled for `harfbuzz`.

I'm going to close this issue, feel free to reopen it if this doesn't work for you. 

I'd accept a PR relating to additional validation of features so that the build script could display a diagnostic if you require `harfbuzz[icu]` but only `harfbuzz` is installed. You would have to detect the condition from `build.rs` and print a message. (Typically, packages that are using `vcpkg` have another way of finding the libraries to link to and the use of `vcpkg` is secondary - for these libraries, if `vcpkg` does not find the library, it just does nothing, assuming that the other mechanism is going to work out. The failure is then an error in the link phase if the library is not found some other way. There is nowhere to display a message unless you fail the build from build.rs, so there might not be anywhere to print a message that will be all that useful.)

@ras0219-msft The install-if-missing thing is a great idea - definitely something that I am interested in. It would require an opt-in using an environment variable I suppose - it seems rude or at least surprising to update someone's vcpkg installation without intervention.
